### DISCUSSION TITLE
Spaces before column names in header row cause HeaderValidationException

### DISCUSSION THREAD

**Discussion by rickhodder** [JoshClose/CsvHelper#1496] [Q&A]
When there are spaces before the beginning of column names in the header row, a HeaderValidationException is thrown, saying that the columns dont exist.

**To Reproduce**

public class MyRow
{
   public string Foo {get;set;}
   public string Bar {get;set;}
   public string Example {get;set;}
}

File contents: headerdemo.txt:

Foo, Bar, Example
F,B,E

Code Sample
using (var reader = new StreamReader(@".\headerdemo.txt"))
            using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
            {                
                var records = csv.GetRecords<MyRow>().ToList();
                return records;
            }

1. When GetRecords runs, this throws a HeaderValidationException saying that Bar cannot be found.
2. When the space is removed before Bar in the header row, this throws a HeaderValidationException saying that Example cannot be found:
Foo,Bar, Example
3. When the space is removed before Example this runs fine with no error:
Foo,Bar,Example

**Expected behavior**
To ignore initial spaces after commas in header row

**Comment by JoshClose** [comment]
Why are there spaces in the header fields and only after the first field? Is this just because you did a hand written example or are you getting files like this?

You can either specify the name with a space in it.

```cs
public class MyRow
{
   public string Foo {get;set;}
   [Name(" Bar")]
   public string Bar {get;set;}
   [Name(" Example")]
   public string Example {get;set;}
}
```

Or you can trim the spaces from the header when matching against your property names.

```cs
csv.Configuration.PrepareHeaderForMatch = (header, index) => header.Trim();
```
